### PR TITLE
Add react-jss example

### PR DESCRIPTION
**Pull request by HenriBeck**
Added a clear example on how to use react-jss with injecting the styles on the server.

cssinjs/jss#457

### CODE REVIEW

**Comment by timneutkens** [comment]
@kof @oliviertassinari could you have a look at this example 🕵️ If it looks alright I'll merge it 👍

**Comment by oliviertassinari** [comment]
@HenriBeck How do you solve the hot reloading class name missmatch issue? On Material-UI we had to remove the moduleId.

**Comment by HenriBeck** [comment]
There aren't any mismatches when hot reloading. I can change the styles or a component, and I get the new changes applied.

I think the reason it works, is because `react-jss` doesn't implement anything specific for HMR updates and remounts the component/tree.

**Comment by oliviertassinari** [comment]
It's good to know.

**Comment by HenriBeck** [comment]
A mismatch should only happen when the client is being hydrated which only happens on the first render if I'm correct. The HMR should only rerender/update the components.

File: examples/with-react-jss/.babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["next/babel"]
```
Comment (timneutkens) [inline examples/with-react-jss/.babelrc]: This seems unneeded 🤔

Comment (HenriBeck) [reply examples/with-react-jss/.babelrc]: I get this error when I don't have this file: 
```sh
Module build failed: Error: [BABEL] /Users/henri/Coding/next.js/examples/with-react-jss/pages/_app.js: You gave us a visitor for the node type PrivateName but it's not a valid type
```

**Review by timneutkens — APPROVED** [review]

File: examples/with-react-jss/.babelrc (hunk 1 above)
Comment (timneutkens) [reply examples/with-react-jss/.babelrc]: You possibly have a `.babelrc` in an upward directory 🤔

Comment (HenriBeck) [reply examples/with-react-jss/.babelrc]: Yeah, that one: https://github.com/zeit/next.js/blob/canary/examples/.babelrc

Comment (timneutkens) [reply examples/with-react-jss/.babelrc]: Ahhh yeah that's fine. Users download only the example itself. Either way it'll work 👍

**Pull request merged**